packages/nova-types/src/nova-localization.ts

```diff
@@ -0,0 +1,29 @@
+const LocalizedString = Symbol("LocalizedString");
+
+export type StringWithPlaceholders<T extends Record<string, string | number>> =
```

Comment:
Let's add docstring for this type. In particular, what are the semantics of the key and value in the `Record` type in the type parameter constraint?

Reply:
In most cases you will not need to use this type the compiler will generate the usage, in which case the type parameter would be pretty clear. That said, I'll add some words on how it's used.